core/stateful_precompile_test.go

```diff
@@ -363,31 +368,23 @@ func TestContractNativeMinterRun(t *testing.T) {
 			},
 			suppliedGas: precompile.ModifyAllowListGasCost,
 			readOnly:    true,
-			setupState: func(state *state.StateDB) {
-				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
-				precompile.SetContractNativeMinterStatus(state, noRoleAddr, precompile.AllowListEnabled)
-			},
-			expectedErr: precompile.ErrWriteProtection.Error(),
+			expectedErr: vmerrs.ErrWriteProtection.Error(),
 		},
-		"mint with insufficient gas": {
-			caller:         noRoleAddr,
+		"readOnly mint with allow role fails": {
```

Comment:
this does not test minting, should we rename this to something like
```suggestion
		"readOnly modify allow list with allow role in mint contract fails": {
```